src/examples/libpmemobj/btree.c

```diff
@@ -48,7 +53,7 @@ POBJ_LAYOUT_END(btree);
 struct btree_node {
 	int64_t key;
 	TOID(struct btree_node) slots[2];
-	char value[];
+	char value[1];
```

Comment:
This is not semantically equivalent. You just created an off-by-one error in the allocation routine.

This type of change should not be made lightly because it's 's very error-prone as you just demonstrated. Please try to find a different solution to this.